framework/src/main/java/org/checkerframework/common/basetype/BaseTypeChecker.java

```diff
@@ -111,7 +116,7 @@ public void initChecker() {
      * list before getSubcheckers() is called, thereby ensuring that this list is non-empty only for
      * one checker.
      */
-    private List<BaseTypeChecker> subcheckers = null;
+    private List<BaseTypeChecker> subcheckers;
```

Comment:
Many of the fields in this and many other classes are not initialized in the constructor, but in helper methods like `initChecker`. We don't initialize these fields with `null`. That would only add redundancy to the field declaration.
This was the only field that has such a redundant initialization, so I fixed it.

Also, standard Java practices advise against doing this:
http://www.javapractices.com/topic/TopicAction.do?Id=14

Reply:
I disagree that omitting initializers is a standard, and it isn't standard in our code -- we don't omit all `= false` initializers for boolean fields.
Also, your discussion of `initChecker` does not seem relevant to this field.
I suggest that we focus on the merits of the design rather than which one is more popular or remarks that relate to other fields.

I gave an argument that an explicit initializer conveys information and is not redundant.  You can also see https://homes.cs.washington.edu/~mernst/advice/coding-style.html#initialization.
In this code review, I wasted time trying to find the initialization after you deleted the clear `= null` that already existed.

I think this should be changed back to good style that explicitly indicates the initial value of a field, unless it is set in the constructor or an initializer block.

If you want to make an unrelated change like this, it should be in a separate pull request, with a discussion of style.

Reply:
We started a separate discussion of this larger issue.
In this case, the change makes the field consistent with the other fields in this class and this PR touched that field.

We could automate checking of this style using
https://chromium-review.googlesource.com/c/chromium/src/+/1209547/41/tools/android/errorprone_plugin/src/org/chromium/tools/errorprone/plugin/NoRedundantFieldInitCheck.java

Reply:
Per my comment, please put this in a separate pull request, since it is unrelated to the other changes in this file.

Reply:
I've created https://github.com/typetools/checker-framework/pull/2956